### PR TITLE
feat(template): Add AlgoliaSearch template

### PR DESCRIPTION
**Pull request by francoischalifour**
## Preview

image

### CODE REVIEW

File: templates/AlgoliaSearch/src/app.js.hbs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import algoliasearch from 'algoliasearch';
+
+const client = algoliasearch('{{appId}}', '{{apiKey}}');
+const index = client.initIndex('{{indexName}}');
+
+const $searchBox = document.querySelector('#searchBox input[type=search]');
```
Comment (Haroenv) [inline templates/AlgoliaSearch/src/app.js.hbs]: the `$` threw me off and I thought it was jQuery, but it isn't

File: templates/AlgoliaSearch/src/app.js.hbs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const $hits = document.querySelector('#hits');
+
+function renderHits(query) {
+  index.search(query).then(result => {
+    $hits.innerHTML = `
+      <ul class="ais-hits">
+        ${result.hits
+          .map(hit => `<li class="ais-hits--item">{{#if mainAttribute}}${hit._highlightResult.{{mainAttribute}}.value}{{else}}${JSON.stringify(hit)}{{/if}}</li>`)
```
Comment (Haroenv) [inline templates/AlgoliaSearch/src/app.js.hbs]: I'd add a comment here that this is vulnerable to XSS, and if people have user-provided data it should be sanitized

Comment (francoischalifour) [reply templates/AlgoliaSearch/src/app.js.hbs]: Because of the `innerHTML`?

File: templates/AlgoliaSearch/src/app.js.hbs (hunk 1 above)
Comment (Haroenv) [reply templates/AlgoliaSearch/src/app.js.hbs]: I knew about the convention, but only saw it commonly used for jquery elements

File: templates/AlgoliaSearch/src/app.js.hbs (hunk 2 above)
Comment (Haroenv) [reply templates/AlgoliaSearch/src/app.js.hbs]: yes, innerHTML leaves this vulnerable, just a comment would be enough warning IMO

**Comment by francoischalifour** [comment]
Closing because we might not want to support the JavaScript client.

**Pull request closed without merging**